lib/semantics/symbol.h

```diff
@@ -78,12 +78,15 @@ class SubprogramDetails {
   const std::vector<Symbol *> &dummyArgs() const { return dummyArgs_; }
   void add_dummyArg(Symbol &symbol) { dummyArgs_.push_back(&symbol); }
   void add_alternateReturn() { dummyArgs_.push_back(nullptr); }
+  const MaybeExpr &stmtFunction() const { return stmtFunction_; }
+  void set_stmtFunction(SomeExpr &&expr) { stmtFunction_ = std::move(expr); }
```

Comment:
Is this used for anything yet?

Reply:
No.  The CG will need it, though.  And I suppose that folding could use it for statement functions in which each dummy argument appears at most once in the body (or whose actual argument can be safely re-evaluated).

Reply:
Also, the presence of the statement function's expression is the only distinguishing characteristic by which semantics could determine that a subprogram is a statement function, and exclude it from use in various circumstances (e.g., passing the statement function as an actual argument, using the statement function in a generic interface, using the statement function as the target of a pointer assignment, &c.) which I'm sure that we still need to do.